qt/python/mantidqt/widgets/codeeditor/completion.py

```diff
@@ -0,0 +1,185 @@
+        # Any default argument appears in the string
+        # on the rhs of an equal
+        for arg in arg_str.split(', '):
+            arg = arg.strip()
+            if '=' in arg:
+                arg_token = arg.split('=')
+                args.append(arg_token[0])
+                defs.append(arg_token[1])
+            else:
+                args.append(arg)
+        if len(defs) == 0:
+            defs = None
+    else:
+        return ''
+
+    if defs is None:
+        call_tip = ', '.join(args)
+        call_tip = '(' + call_tip + ')'
+    else:
+        # The defaults list contains the default values for the last n arguments
+        diff = len(args) - len(defs)
+        call_tip = ''
+        for index in range(len(args) - 1, -1, -1):
+            def_index = index - diff
+            if def_index >= 0:
+                call_tip = '[' + args[index] + '], ' + call_tip
+            else:
+                call_tip = args[index] + ", " + call_tip
+        call_tip = '(' + call_tip.rstrip(', ') + ')'
+    return call_tip
+
+
+def generate_call_tips(definitions, prepend_module_name=None):
+    """
+    Generate call tips for a dictionary of object definitions (eg. globals()).
+    The call tips generated are of the form:
+        Load(InputWorkspace, [OutputWorkspace], ...)
+    where squared braces denote key-word arguments.
+
+    :param dict definitions: Dictionary with names of python objects as keys and
+        the objects themselves as values
+    :param str prepend_module_name: Prepend the name of the module to the call tips
+        default is None
+    :returns list: A list of call tips
+    """
+    if not isinstance(definitions, dict):
+        return []
+    call_tips = []
+    for name, py_object in definitions.items():
+        if name.startswith('_'):
+            continue
+        if inspect.isfunction(py_object) or inspect.isbuiltin(py_object):
+            if not prepend_module_name:
+                call_tips.append(name + get_function_spec(py_object))
+            else:
+                call_tips.append(prepend_module_name + '.' + name + get_function_spec(py_object))
+            continue
+        # Ignore modules or we get duplicates of methods/classes that are imported
+        # in outer scopes, e.g. numpy.array and numpy.core.array
+        if inspect.ismodule(py_object):
+            continue
+        for attr in dir(py_object):
+            try:
+                f_attr = getattr(py_object, attr)
+            except Exception:
+                continue
+            if attr.startswith('_'):
+                continue
+            if hasattr(f_attr, 'im_func') or inspect.isfunction(f_attr) or inspect.ismethod(f_attr):
+                call_tip = name + '.' + attr + get_function_spec(f_attr)
+            else:
+                call_tip = name + '.' + attr
+            if prepend_module_name:
+                call_tips.append(prepend_module_name + '.' + call_tip)
+    return call_tips
+
+
+class CodeCompleter:
```

Comment:
As we're not out of Python 2 (just yet) can be inherit from object so we know everything is a new-style class?